spec/src/main/asciidoc/WebSocket.adoc

```diff
@@ -1291,11 +1291,35 @@ been implemented as a *Filter*. If the WebSocket upgrade process has not been
 implemented as a *Filter*, additional container specific configuration may be
 required.
 
-The position of the WebSocket upgrade *Filter* in the *FilterChain* is
-configured using the standard Filter ordering rules of the Servlet
-specification. *Filter* instances earlier in the *FilterChain* may wrap the
-request and/or response and these wrappers will be visible to, and may
-influence, the WebSocket upgrade *Filter*.
+The WebSocket upgrade *Filter* is positioned in the *FilterChain* after any
+**Filter**s defined in the deployment descriptor (*web.xml*). Applications
+requiring that the WebSocket upgrade *Filter* is positioned earlier in the
+*FilterChain* may define it explicitly in the deployment descriptor.
+
+*Filter* instances earlier in the *FilterChain* may manipulate and/or wrap the
+request and/or response and these behaviours may influence the functionality of
+the WebSocket upgrade *Filter*. Developers should be aware that some *Filter*
+implementations may not have considered HTTP upgrade and configuring such
+**Filter**s before the WebSocket upgrade *Filter* in the *FilterChain* may break
+the HTTP upgrade process.
+
+If the response passed to the *doFilter()* method of the WebSocket upgrade
+*Filter* has been committed, then the WebSocket upgrade *Filter* will take no
+further action and pass the request and response to the next *Filter* in the
+*FilterChain*.
+
+If the response passed to the *doFilter()* method of the WebSocket upgrade
+*Filter* has not been committed and the WebSocket upgrade *Filter* determines
+that the criteria have been met to allow the upgrade to WebSocket to proceed,
+then the response will be reset before the WebSocket upgrade *Filter* writes the
+appropriate response headers and status code.
+
+Applications that require a greater degree of control over the HTTP upgrade
+process should use *WsServerContainer.upgradeHttpToWebSocket* method to manually
+trigger the HTTP upgrade process to WebSocket. Such applications will probably
+wish to avoid the use of WebSocket annotations so that the container does not
+automatically deploy WebSocket endpoints as described in
+<<application-deployment-on-web-containers>>.
```

Comment:
Can we get a ServletContext attribute that disables/configures specific jakarta.websocket features?

1. WebSocket Upgrade as a Filter from implementation (options: enable/disable, default is enabled)
2. WebSocket Upgrade as a Filter location (options: before/after deployment descriptor from webapp, default is before)

I could see some attributes to disable automatic endpoint additions too.
This would be useful for some scenarios.
There's also the scanning of the webapp to support `jakarta.websocket.server.ServerApplicationConfig` features.

1. Scan and deploy annotated endpoints.
2. Scan and report to ServletApplicationConfig entries.